### PR TITLE
[WIP] Example of RAJA interoperability [artv3/interop-raja] 

### PR DESCRIPTION
**Pull request by artv3**
Hi Folks, building on PR #937 . This example illustrates interoperability between RAJA and MFEM's memory manager. Additionally, data is allocated externally and then passed into an MFEM vector. I show how to operate on the data on both the GPU and CPU (using RAJA) and maintain validity of the data flags. I found this exercise useful and thought it might be good to share.

### CODE REVIEW

File: examples/main.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+   double * xsub = &x[nSub];   //Pointer to subvector data
+
+   //Poulate base vector on the host
+   for(int i=0; i<N; ++i) x[i] = i+1;
+
+   //Shallow copy into MFEM vector + read command to copy to the GPU
+   Vector V(x, N); V.Read();
+
+   //Shallow copy into MFEM vector + annotate as reference to a base vector
+   Vector Vs(xsub, nSub); Vs.MakeRef(V, nSub);
```
Comment (artv3) [inline examples/main.cpp]: Alternatively, create an empty vector then use MakeRef(vector, offset, size)

File: examples/main.cpp [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+   //Sync validity flags with base Vector
+   Vs.SyncAliasMemory(V);
```
Comment (artv3) [inline examples/main.cpp]: Not needed...

File: examples/main.cpp [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+   //Modify data on host
+   RAJA::forall<RAJA::loop_exec>
+     (RAJA::RangeSegment(0,nSub), [=]  (int i) {
+       h_xs[i] = 777.0;
+     });
+
+   //Sync validity flags with sub vector
+   V.SyncMemory(Vs);
+
```
Comment (artv3) [inline examples/main.cpp]: prefer Vs.SyncAliasMemory(v), won't need read at the bottom (but doesn't hurt)

**Comment by tzanio** [comment]
@artv3 -- is this something you will like to merge eventually or is it just an example that could be useful for other users? In the latter case, I think it will be better to preserve it in an issue...

**Comment by artv3** [comment]
@tzanio , this was mostly an exercise to get a feel for how the memory manager worked. We don't have to merge this. I think it could be useful for application developers. If you like, I can do a cleanup pass and we can preserve it in an issue.

**Comment by tzanio** [comment]
> If you like, I can do a cleanup pass and we can preserve it in an issue.
Yes, please do that if you don't mind. I agree that it will be useful for other users.

**Comment by artv3** [comment]
Closing/Deleting this PR. Will move information to an MFEM issue this week.

**Pull request closed without merging**